File: src/main/java/org/grandviewtech/constants/InputType.java
package org.grandviewtech.constants;

public enum InputType
	{
	INPUT("INPUT"), FLAG("FLAG"), WORD("WORD"), OUTPUT("OUTPUT");
		
		private String inputType;
		
		private InputType(String inputType)
			{
				this.inputType = inputType;
			}
			
		public String getInputType()
			{
				return inputType;
			}
	}

File: ladder-core/crates/ladder-model/src/input_type.rs
use serde::{Deserialize, Serialize};

/// Ported from org.grandviewtech.entity.enums.InputType.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum InputType {
    Input,
    Flag,
    Word,
    Output,
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn wire_form_matches_original_java_enum_strings() {
        assert_eq!(serde_json::to_string(&InputType::Flag).unwrap(), "\"FLAG\"");
        assert_eq!(serde_json::to_string(&InputType::Word).unwrap(), "\"WORD\"");
    }
}
